incubator/hnc/pkg/forest/forest.go

```diff
@@ -100,16 +101,8 @@ func (ns *Namespace) clean() {
 // children of its own parent, if necessary. It may return an error if the parent is illegal, i.e.
 // if it causes a cycle. It cannot cause an error if the parent is being set to nil.
 func (ns *Namespace) SetParent(p *Namespace) error {
-	// Check for cycles
-	if p != nil {
-		// Simple case
-		if p == ns {
-			return fmt.Errorf("%q cannot be set as its own parent", p.name)
-		}
-		if chain := p.AncestoryNames(ns); chain != nil {
-			return fmt.Errorf("cycle when making %q the parent of %q: current ancestry is %q",
-				p.name, ns.name, strings.Join(chain, " <- "))
-		}
+	if reason := ns.CanSetParent(p); reason != "" {
```

Comment:
reason -> err?

Reply:
I didn't really want to return an error, since we're not really requesting an operation here. It's more like a dry run. I can change it if you feel strongly though.

Reply:
Effectively, a non empty reason means something is wrong, hence the suggestion. Not a strong opinion though.